packages/core/admin/admin/src/content-manager/components/DynamicZone/components/Component/util.js

```diff
@@ -0,0 +1,16 @@
+import { useMemo } from 'react';
+import { get, toString } from 'lodash';
+import { useCMEditViewDataManager } from '@strapi/helper-plugin';
+
+function useMainValue(schema, componentFieldName) {
```

Comment:
Could you add a test for this file?

Reply:
I think the best would be to copy this file into a `utils` folder (e.g. `./utils/useMainValue.js`. Then you can add a `test` folder within `utils` and write a basic unit-test similar to [this one](https://github.com/strapi/strapi/blob/5c5ef16451c641f5970114cc99763635ce3591a7/packages/core/admin/admin/src/content-manager/components/DynamicTable/CellContent/utils/tests/hasContent.test.js), which tests input / output.

Reply:
## Run tests - not working with yarn test*

> I have tried to run the unit tests based on the main (monorepo) and the core/admin package.json all test scripts (), without success.

How could I run a single test file? 

```
(base) PS strapi\packages\core\admin> yarn test:front
yarn run v1.22.17
$ cross-env IS_EE=true jest --config ./jest.config.front.js
No tests found, exiting with code 1
Run with `--passWithNoTests` to exit with code 0
In strapi
  1108 files checked.
  testMatch: /**/tests/**/?(*.)+(spec|test).[jt]s?(x) - 0 matches
  testPathIgnorePatterns: \\node_modules\\, strapi\\examples\\getstarted\\, strapi\\examples\\kitchensink\\, strapi\\packages\\strapi-helper-plugin\\dist\\, __tests__ - 1080 matches
  testRegex:  - 0 matches
Pattern:  - 0 matches
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
(base) PS strapi\packages\core\admin>
```

## Mocking `useCMEditViewDataManager` data

The hasContent is not a hook, and has simple parameters (arguments) and a simple logic which is easily unit testable.

> Or should I only test the `let displayedValue = toString(get(modifiedData, [...componentFieldPath, mainField], ''));` expression?

The useMainValue is a hook with usage of other Hooks (useCMEditViewDataManager, useMemo) with complex values (_shoud be mocked somehow_).

I found this only two example ( [EditView/Informations](https://github.com/godzzo/strapi/blob/b990a4227bb4721ad12f4da4c55ed4e152ffdaad/packages/core/admin/admin/src/content-manager/pages/EditView/Informations/tests/index.test.js#L44), [InputUID](https://github.com/godzzo/strapi/blob/e0e415551b0c38771f1c9aa580503abd7b38a456/packages/core/admin/admin/src/content-manager/components/InputUID/tests/index.test.js#L15) ) without calling the mock explicitly, and I coud not run them, that could be a good example for the `useMainValue` hook?

```js
jest.mock('@strapi/helper-plugin', () => ({
	...jest.requireActual('@strapi/helper-plugin'),
	useCMEditViewDataManager: jest.fn(() => ({
		modifiedData: {},
		initialData: {},
	})),
}));
```

Reply:
I have decoupled the `displayedValue` logic and pushed a unit test file for it, but could not run the test.